File: Formula-Bollo/src/shared/models/fantasyPriceTeam.ts
import { Race } from "./race";
import { Season } from "./season";
import { Team } from "./team";

export class FantasyPriceTeam {
  id: number;
  team: Team;
  race: Race;
  price: number;
  season: Season;
  totalPoints: number = 0;
  averagePoints: number = 0;
  differencePrice?: number;

  constructor(
    id: number,
    team: Team,
    race: Race,
    price: number,
    season: Season,
  ) {
    this.id = id;
    this.team = team;
    this.race = race;
    this.price = price;
    this.season = season;
  }
}

File: Formula-Bollo-App/src/main/java/formula/bollo/app/entity/FantasyPriceTeam.java
package formula.bollo.app.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Entity
@Table(name = "fantasy_price_team")
@Getter
@Setter
@ToString
@EqualsAndHashCode
@AllArgsConstructor
@NoArgsConstructor
public class FantasyPriceTeam {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @ManyToOne
    @JoinColumn(name = "team_id", referencedColumnName = "id")
    private Team team;

    @ManyToOne
    @JoinColumn(name = "race_id", referencedColumnName = "id")
    private Race race;

    @Column(name = "price", nullable = false)
    private int price;

    @ManyToOne
    @JoinColumn(name = "season_id", referencedColumnName = "id")
    private Season season;
}
